utils/data-file.c

```diff
@@ -255,15 +255,15 @@ int read_events_file(struct uftrace_data *handle)
 		struct uftrace_event *ev;
 
 		if (!strncmp(line, "EVENT", 5)) {
-			sscanf(line + 7, "%u %[^:]:%s",
-			       &evt_id, provider, event);
+			if (sscanf(line + 7, "%u %[^:]:%s", &evt_id, provider, event) != 3) {
```

Comment:
Shouldn't it be `sscanf(...) == 3`?

In fact, I think this looks better as it touches less code:
```
diff --git a/utils/data-file.c b/utils/data-file.c
index d01580dbe2..b2c1f4180b 100644
--- a/utils/data-file.c
+++ b/utils/data-file.c
@@ -254,9 +254,9 @@ int read_events_file(struct uftrace_data *handle)
                unsigned evt_id;
                struct uftrace_event *ev;
 
-               if (!strncmp(line, "EVENT", 5)) {
-                       sscanf(line + 7, "%u %[^:]:%s",
-                              &evt_id, provider, event);
+               if (!strncmp(line, "EVENT", 5) &&
+                   sscanf(line + 7, "%u %[^:]:%s",
+                          &evt_id, provider, event) == 3) {
 
                        ev = xmalloc(sizeof(*ev));
                        ev->id = evt_id;
```